### PR TITLE
Fixes #625: Add YETI Connector

### LINKED ISSUE #625: [New Connector] YETI
I see a scope for adding a `YETI Connector` as well, like #12.

_Originally posted by @sp35 in https://github.com/intelowlproject/IntelOwl/issues/426#issuecomment-824199572_

### PR DESCRIPTION
**Pull request by sp35**
# Description
Adds connector for YETI platform

## Related issues
#625 

## Type of change

- [x] New feature (non-breaking change which adds functionality).

# Checklist

- [x] The pull request is for the branch develop
- [x] If I added a new analyzer, I updated the file [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md). If the analyzer provides additional optional configuration, I added the available options here: [Advanced-Usage](./Advanced-Usage.md)
- [x] If I added external libraries/packages that use restrictive licenses, please add them in the [ReadMe - Legal Notice](https://github.com/certego/IntelOwl/blob/master/README.md) section
- [x] I added new secrets in the files [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the docs: [Installation](./Installation.md)
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [ ] I squashed the commits into a single one. (optional, they will be squashed anyway by the maintainer)

# Real World Example
Screenshot from 2021-08-12 13-28-59

### CODE REVIEW

File: api_app/connectors_manager/connectors/yeti.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        # get job tags
+        tags = list(self._job.tags.all().values_list("label", flat=True))
+
+        # create observable with `obs_value` if it doesn't exists
+        # new context, tags, source are appended with existing ones
+        result = self.yeti_instance.observable_add(
+            value=obs_value, tags=tags, context=context, source="IntelOwl"
+        )
+
+        if result is None:
```
Comment (sp35) [inline api_app/connectors_manager/connectors/yeti.py:47]: https://github.com/intelowlproject/IntelOwl/issues/426#issuecomment-897511057

File: api_app/connectors_manager/connectors/yeti.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+from api_app.exceptions import ConnectorRunException
+from django.conf import settings
+
+import pyeti
```
Comment (Eshaan7) [inline api_app/connectors_manager/connectors/yeti.py:7]: nitpick: move this import to top.

Comment (Eshaan7) [reply api_app/connectors_manager/connectors/yeti.py:7]: and keep the `api_app` imports together.

File: configuration/connector_config.json [hunk 3]
```diff
@@ -61,5 +61,31 @@
         "description": "URL of your OpenCTI instance"
       }
     }
+  },
+  "YETI": {
+    "disabled": true,
+    "description": "finds/create an observable on your YETI instance, linking the the successful analysis on IntelOwl",
```
Comment (Eshaan7) [inline configuration/connector_config.json:67]:
```
find or create an observable on an YETI instance, linking the successful analysis on IntelOwl.
```

File: docs/source/Usage.md [hunk 4]
```diff
@@ -194,6 +194,7 @@ You can programmatically retrieve all the available connectors with the official
 The following is the list of the available connectors:
 * `MISP`: automatically creates an event on your MISP instance, linking the successful analysis on IntelOwl.
 * `OpenCTI`: automatically creates an observable and a linked report on your OpenCTI instance, linking the the successful analysis on IntelOwl.
+* `YETI` (Your Everyday Threat Intelligence): finds/create an observable on your YETI instance, linking the the successful analysis on IntelOwl.
```
Comment (Eshaan7) [inline docs/source/Usage.md:197]: change here too.

File: api_app/connectors_manager/connectors/yeti.py (hunk 2 above)
Comment (sp35) [reply api_app/connectors_manager/connectors/yeti.py:7]: sometimes I forget to check auto imports :sweat_smile:

**Pull request merged**